prusti-server/src/verifier_runner.rs

```diff
@@ -0,0 +1,74 @@
+// © 2020, ETH Zurich
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use prusti_common::{
+    config,
+    report::log,
+    vir::{Program, ToViper},
+    Stopwatch,
+};
+use prusti_viper::{
+    verification_service::ViperBackendConfig,
+    verifier::{VerificationContext, VerifierBuilder},
+};
+use viper::{self, VerificationResult};
+
+pub struct VerifierRunner<'v> {
+    verifier: viper::Verifier<'v, viper::state::Started>,
+    ast_factory: viper::AstFactory<'v>,
+    ast_utils: viper::AstUtils<'v>,
+}
+
+impl<'v> VerifierRunner<'v> {
+    pub fn with_runner<F, Res>(
+        verifier_builder: &VerifierBuilder,
+        backend_config: &ViperBackendConfig,
+        body: F,
+    ) -> Res
+    where
+        F: FnOnce(VerifierRunner) -> Res,
+    {
+        let context = verifier_builder.new_verification_context();
+        let runner = VerifierRunner::new(&context, backend_config);
+        body(runner)
+    }
+
+    pub fn with_default_configured_runner<F, Res>(
+        verifier_builder: &VerifierBuilder,
+        body: F,
+    ) -> Res
+    where
+        F: FnOnce(VerifierRunner) -> Res,
+    {
+        Self::with_runner(verifier_builder, &ViperBackendConfig::default(), body)
+    }
+
+    fn new(context: &'v VerificationContext, backend_config: &ViperBackendConfig) -> Self {
+        Self {
+            verifier: context.new_viper_verifier(backend_config),
+            ast_factory: context.new_ast_factory(),
+            ast_utils: context.new_ast_utils(),
+        }
+    }
+
+    pub fn verify(&self, program: Program, program_name: &str) -> VerificationResult {
+        let mut stopwatch = Stopwatch::start("construction of JVM objects");
+        let viper_program = program.to_viper(&self.ast_factory);
+        if config::dump_viper_program() {
+            self.dump(viper_program, program_name);
+        }
+
+        stopwatch.start_next_section("verification");
+        self.verifier.verify(viper_program)
+    }
```

Comment:
Could you measure and log the duration of `to_viper`, `dump` and `verify` with the `Stopwatch`?